gocdpasswd/src/main/java/cd/go/authentication/crypt/hash/BCryptProvider.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cd.go.authentication.crypt.hash;
+
+import cd.go.authentication.crypt.CliArguments;
+import org.mindrot.jbcrypt.BCrypt;
+
+import static java.text.MessageFormat.format;
+
+public class BCryptProvider implements HashProvider {
+
+    @Override
+    public String hash(CliArguments arguments) {
+        if (arguments.cost() > 15) {
```

Comment:
This is arbitrary. On a slow computer, even an argument of 5 may make forever. Instead, just say the statement without the surrounding if clause.